main.go

```diff
@@ -0,0 +1,179 @@
+package main
+
+import (
+	"context"
+	"sync"
+	"time"
+
+	"github.com/aws/aws-sdk-go/aws"
+	"github.com/aws/aws-sdk-go/aws/session"
+	"github.com/aws/aws-sdk-go/service/s3"
+	"github.com/aws/aws-sdk-go/service/s3/s3manager"
```

Comment:
Not that important, but we might as well use the https://github.com/aws/aws-sdk-go-v2 ?